src/context/product-provider.js

```diff
@@ -0,0 +1,33 @@
+import {
+    useContext,
+    createContext,
+    useState,
+    useEffect
+  } from "react";
+  import axios from "axios";
+  
+  const ProductContext = createContext();
+  const useProduct = () => useContext(ProductContext);
+  
+  const ProductProvider = ({ children }) => {
+
+    const [productsList, setProductsList] = useState([]);
+    const productApiCall = async () => {
+      const response = await axios.get("/api/products");
```

Comment:
try-catch should be added for error handling of API fetching

Reply:
i will do it in next pr